### ISSUE TITLE
Error in creating nested `ArrayNode`s with `JsonNode.withArray()`

### ISSUE THREAD

**Issue by SaiKrishna369** [FasterXML/jackson-databind#3882]
**Describe the bug**
Attempt to create a new path to an ArrayNode with `withArray` method where the pointer would include an array index in between is failing with `java.lang.UnsupportedOperationException`.

**Version information**
2.14.2

**To Reproduce**
```
@Test
public static void test_nestedArray() {
        ObjectMapper mapper = new ObjectMapper();
        String json = "{}";
        try {
            JsonNode jsonObject = mapper.readTree(json);
            ArrayNode aN = jsonObject.withArray("/key1/array1/0/element1", JsonNode.OverwriteMode.ALL, true);
            aN.add("v1");
            aN.add("v2");
            aN.add("v3");
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }
```
Executing this method is throwing:
```
java.lang.UnsupportedOperationException: Cannot replace context node (of type `com.fasterxml.jackson.databind.node.ObjectNode`) using `withArray()` with  JSON Pointer '/key1/array1/0/element1'
	at com.fasterxml.jackson.databind.node.BaseJsonNode._reportWrongNodeType(BaseJsonNode.java:257)
	at com.fasterxml.jackson.databind.node.BaseJsonNode.withArray(BaseJsonNode.java:193)
	at com.fasterxml.jackson.databind.JsonNode.withArray(JsonNode.java:1319)
```

**Expected behavior**
A new path with `key1` as object, containing `array1` whose first element is an object containing an array with name `element1` should be created, i.e. the resultant json should be`{"key1": { "array1": [ {"element1": ["v1", "v2", "v3"] } ] } }`.

**Additional context**
`ArrayNode aN = jsonObject.withArray("/key1/array1/element1", JsonNode.OverwriteMode.ALL, true)` and `ObjectNode oN = jsonObject.withObject("/key1/array1/0/element1", JsonNode.OverwriteMode.ALL, true)` are working as expected.

**Comment by pjfanning** [comment]
As the exception message says, this is unsupported. You cannot use withArray() to convert an ObjectNode. If you want to change the behaviour, maybe you could look at submitting a PR?

**Comment by cowtowncoder** [comment]
I'd have to double-check this, but it might be a bug. With `OverwriteMode.ALL` it should be legal to replace the last node.
But I may have missed something.

**Comment by cowtowncoder** [comment]
Yup, a bug indeed. Fixed in 2.14 branch, but most likely released next in 2.15.1 release.
